### PR TITLE
[jquery] Clean up event-related methods on JQuery and fix issues with importing

### PR DESCRIPTION
**Pull request by leonard-thieu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- ~~Provide a URL to documentation or source code which provides context for the suggested changes:~~
- ~~Increase the version number in the header if appropriate.~~
- ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~~

**Summary of changes**
- Add README describing usage and guidance for authoring jQuery plugin typedefs
- Importing
  - The factory function is now declared on `JQueryStatic` with an extraneous `discriminator` parameter
    - Exporting `typeof factory & JQueryStatic` results in type assertions not working on the imported `JQueryStatic` object 
(related: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/844920fa549d0c74b5ac2d2d14858a3674952383#commitcomment-22681634)
    - The `discriminator` parameter is needed to differentiate between `factory(Window)` and `JQueryStatic(Window)`. It does not actually exist on `factory`.
    - I'd much prefer a cleaner, less hackier solution
- Events
  - Overloads that accepted event data **and** `false` for `handler` were removed
  - `bind(eventType, handler)` now accepts `null` and `undefined` since jQuery accepts those values for compatibility reasons
  - Fix `on()` and `one()` overloads to reflect that passing `string` as data does not work
  - Fix `trigger()` and `triggerHandler()` to allow `string` and `number` to be passed to `extraParameters`
  - Change `Event` from a `class` to an `interface`
    - This was incorrectly declaring a value at `JQuery.Event` (should be `jQuery.Event`)
    - This probably requires declaring `JQuery` as a class in order to correctly declare `Event` as an inner class
  - [minor] Methods that accept a selector now specify `JQuery.Selector` as the type instead of `string`
  - [minor] `hover()` now uses the virtual equivalent `JQuery.EventHandler` for its callbacks
  - [minor] Reordered and grouped `off()` overloads to be easier to read (should still be equivalent to what they were before)

---
Would really appreciate some reviews before merging.

### CODE REVIEW

**Comment by leonard-thieu** [comment]
@Kovensky mind taking a look at this?

**Comment by johnnyreilly** [comment]
Thanks!

File: types/jquery/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#### Global
+
+When jQuery is globally available, you can use `jQuery` and `$` directly.
+
+#### Importing (with a global DOM available)
+
+When you want to import jQuery as a module and have a global DOM available (e.g. browser and browser-like environments):
+
+```typescript
+import * as jQuery from 'jquery';
```
Comment (Kovensky) [inline types/jquery/README.md]: Hm... `import jQuery = require('jquery')` is probably a better example.

`import * as jQuery from 'jquery'` is most certainly invalid under the currently proposed commonjs interop spec; but TypeScript doesn't have a better emit for the would-be correct `import jQuery from 'jquery'` yet.

Comment (johnnyreilly) [inline types/jquery/README.md]: Actually with `allowSyntheticDefaultImports` set to true `import jQuery from 'jquery'` is valid. https://www.typescriptlang.org/docs/handbook/compiler-options.html

Comment (Kovensky) [inline types/jquery/README.md]: It will type check, but the emitted code will be wrong unless you are using --modules es2015

Comment (johnnyreilly) [inline types/jquery/README.md]: I always do :smile:

Comment (leonard-thieu) [inline types/jquery/README.md]: Thanks. I'll change it in my next PR.

**Pull request merged**